### PR TITLE
card level themes

### PR DESCRIPTION
**Pull request by iantrich**
weather-forecast
shopping-list
plant-status
markdown
alarm-panel
picture
picture-elements
picture-entity
picture-glance

Docs: https://github.com/home-assistant/home-assistant.io/pull/10912

### CODE REVIEW

File: src/panels/lovelace/cards/hui-alarm-panel-card.ts
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-alarm-panel-card.ts]: We should also check if the theme is changed:
```
    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
    if (!oldHass || oldHass.themes !== this.hass.themes) {
```

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: Ohw and `shouldUpdate` should be updated, it won't update now on theme changes.

Comment (iantrich) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: I didn't know you could add themes on the fly

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: `frontend.reload_themes`

Comment (iantrich) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: 🤦‍♀️
I'll get that add to the other cards as well as I don't think they have that either.

File: src/panels/lovelace/cards/hui-alarm-panel-card.ts
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-alarm-panel-card.ts]: Should also be added to `shouldUpdate`, it will not come here because `shouldUpdate` will return false.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: And we should also act when `this._config.theme` changes

Comment (iantrich) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: so
```yaml
const oldConfig = changedProps.get("_config");
if (!oldHass || !oldConfig || !oldHass.themes !== this.hass.themes || oldConfig.theme !== this._config.theme)
```
?

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-alarm-panel-card.ts]: Yes

File: src/panels/lovelace/cards/hui-picture-glance-card.ts
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-picture-glance-card.ts]: if (!oldHass) {
    return true;
}

File: src/panels/lovelace/common/has-changed.ts
Comment (bramkragten) [inline src/panels/lovelace/common/has-changed.ts]: I think we should add language here too.

**Review by bramkragten — APPROVED** [review]

**Comment by iantrich** [comment]
blocked by https://github.com/home-assistant/home-assistant-polymer/pull/4079
can't verify my editor changes until this is merged and I rebase

**Comment by bramkragten** [comment]
Merged it

**Pull request merged**
